proof.go

```diff
@@ -17,18 +15,61 @@ func (ap *AuthenticationPath) PrunedTree() [][]byte {
 	return ap.prunedHashes
 }
 
-func (ap *AuthenticationPath) Index() []byte {
-	return ap.index
-}
-
 func (ap *AuthenticationPath) LookupIndex() []byte {
 	return ap.lookupIndex
 }
 
-func (ap *AuthenticationPath) Level() int {
-	return ap.level
-}
-
 func (ap *AuthenticationPath) Leaf() ProofNode {
 	return ap.leaf
 }
+
+type ProofNode interface {
+	Level() int
+	Index() []byte
+	Value() []byte
+	IsEmpty() bool
+	Commitment() []byte
+}
```

Comment:
Is it what the client needs to do manually? I think functions in `proof_test.go` are what the client needs to implement when receiving an authentication path, no?